activestorage/app/models/active_storage/attachment.rb

```diff
@@ -22,12 +22,14 @@ class ActiveStorage::Attachment < ActiveRecord::Base
   # Synchronously deletes the attachment and {purges the blob}[rdoc-ref:ActiveStorage::Blob#purge].
   def purge
     delete
+    record&.touch
```

Comment:
I think delete and touch should be wrapped in a transaction here, as they would have been on Rails 5.2 when the touch happened in an `after_destroy` callback.

```suggestion
    transaction do
      delete
      record&.touch
    end
```

Reply:
@eugeneius  Change done on `purge` and `purge_later`.